### PR TITLE
Split reference_datetime to reference_date and reference_time

### PR DESCRIPTION
**Pull request by Meisterschueler**
For import log files it is better if the reference_time is just optional. So I split the datetime to date and time.

### CODE REVIEW

**Comment by dspreitz** [comment]
Here is the Perl script I use to log messages
`#!/usr/bin/perl -w                                                                                                                                                                                                                
use strict;
use Ham::APRS::IS;
use Ham::APRS::FAP qw(parseaprs);
use Data::Dumper;
use warnings;
 
print "connecting to server\n";

# my $is = new Ham::APRS::IS('aprs.glidernet.org:14580', 'PerlEx', 'appid' => 'Perl Example App', 'filter'=>'r/+45.557/+5.976/201.4');
# my $is = new Ham::APRS::IS('aprs.glidernet.org:14580', 'PerlEx', 'appid' => 'Perl Example App', 'filter'=>'r/+48.751171/+11.466587/500');
my $is = new Ham::APRS::IS('glidern3.glidernet.org:14580', 'PerlEx', 'appid' => 'Perl Example App', 'filter'=>'r/+48.751171/+11.466587/500');

$is->connect('retryuntil' => 3) || die "Failed to connect: $is->{error}";
 
my $lastkeepalive = time();
 
while($is->connected()) {
 
    # make sure we send a keep alive every 240 seconds or so                                                                                                                                                                      
    my $now = time();
    if( $now - $lastkeepalive > 240 ) {
        $is->sendline('# example code');
        $lastkeepalive = $now;
    }
 
    # read the line from the server                                                                                                                                                                                               
    my $line = $is->getline();
    next if (!defined $line);
 
    # parse the aprs packet                                                                                                                                                                                                       
    my %packetdata;
    my $retval = parseaprs($line, \%packetdata);
 
    # and display it on the screen                                                                                                                                                                                                
    if ($retval == 1) {
        print "$line\n"; 
	# print Dumper( \%packetdata );
        # say $fh Dumper( \%packetdata );
    }
}

$is->disconnect() || die "Failed to disconnect: $is->{error}";

`

File: ogn/parser/parse.py [hunk 1]
```diff
@@ -6,16 +6,18 @@
 from ogn.parser.exceptions import AprsParseError, OgnParseError
 
 
-def parse_aprs(message, reference_date=None):
+def parse_aprs(message, reference_date=None, reference_time=None):
     if reference_date is None:
```
Comment (kerel-fs) [inline ogn/parser/parse.py]: If `reference_time` is set, but `reference_date == None`, the `reference_time` is ignored.
I would split those checks:
```
if reference_date is None:
    reference_date = datetime.utcnow().date()
if reference_time is None:
    reference_time = datetime.utcnow().time()
```

Comment (Meisterschueler) [inline ogn/parser/parse.py]: If we call createTimestamp(message, date, None) then message gets a timestamp with the date.
If we call createTimestamp(message, date, time) then message gets a timestamp with the corrected date.
So if we set a reference_time from None to utcnow().time the function createTimestamp will make the corrected date. A logfile (25.09.) with a entry at 00:01 imported at 23:59 will create a Timestamp 26.09. 00:01... Not good. Or did I miss the point?

Comment (kerel-fs) [inline ogn/parser/parse.py]: No, you're right. I can't find any case in which calling `parse_aprs` with `date=None` and `time=some_time` makes sense, so the exact behavior of the function in this case could be ignored.

**Pull request merged**